shared/src/commonMain/kotlin/at/asitplus/wallet/app/common/Extentions.kt

```diff
@@ -0,0 +1,124 @@
+    } ?: throw Throwable("Missing Pattern")
+
+    val scheme = AttributeIndex.schemeSet.firstOrNull {
+        when (credentialRepresentation) {
+            ConstantIndex.CredentialRepresentation.PLAIN_JWT -> throw Throwable("PLAIN_JWT not implemented")
+            ConstantIndex.CredentialRepresentation.SD_JWT -> it.sdJwtType == pattern
+            ConstantIndex.CredentialRepresentation.ISO_MDOC -> it.isoDocType == pattern
+        }
+    } ?: throw Throwable("Missing scheme")
+
+    val constraintsMap = InputEvaluator().evaluateConstraintFieldMatches(
+        inputDescriptor = this,
+        credential = scheme.toJsonElement(credentialRepresentation),
+        pathAuthorizationValidator = { true },
+    ).getOrNull() ?: throw Throwable("Unable to evaluate constraints")
+
+    val attributes = constraintsMap.mapNotNull {
+        val path = it.value.map { value ->
+            value.normalizedJsonPath
+        }.firstOrNull() ?: return@mapNotNull null
+        val optional = it.key.optional ?: return@mapNotNull null
+        path to optional
+    }.toMap()
+
+    return Triple(credentialRepresentation, scheme, attributes)
+}
+
+fun ConstantIndex.CredentialScheme.toJsonElement(
+    representation: ConstantIndex.CredentialRepresentation,
+): JsonElement {
+    val dataElements = when (this) {
+        ConstantIndex.AtomicAttribute2023, IdAustriaScheme, EuPidScheme, MobileDrivingLicenceScheme, HealthIdScheme -> this.claimNames
+        PowerOfRepresentationScheme -> PowerOfRepresentationDataElements.ALL_ELEMENTS
+        CertificateOfResidenceScheme -> CertificateOfResidenceDataElements.ALL_ELEMENTS
+        CompanyRegistrationScheme -> CompanyRegistrationDataElements.ALL_ELEMENTS
+        TaxIdScheme -> TaxIdScheme.ALL_ELEMENTS
+        else -> TODO("${this::class.simpleName} not implemented in jsonElementBuilder yet")
+    }
+
+    return dataElements.associateWith { "" }.let { attributes ->
+        if (representation != ConstantIndex.CredentialRepresentation.ISO_MDOC) {
+            val representationConstraint = when (representation) {
+                ConstantIndex.CredentialRepresentation.PLAIN_JWT -> "type" to this.vcType
+                ConstantIndex.CredentialRepresentation.SD_JWT -> "vct" to this.sdJwtType
+                ConstantIndex.CredentialRepresentation.ISO_MDOC -> "" to this.isoDocType //case can never happen
+            }
+            vckJsonSerializer.encodeToJsonElement(attributes + representationConstraint)
+        } else {
+            vckJsonSerializer.encodeToJsonElement(mapOf(this.isoNamespace to attributes))
+        }
+    }
+}
+
+fun SubjectCredentialStore.StoreEntry.getAttributes(): Map<NormalizedJsonPath, String> = when (val credential = this) {
+    is SubjectCredentialStore.StoreEntry.Vc -> {
+        throw Throwable("Not implemented")
+    }
+
+    is SubjectCredentialStore.StoreEntry.SdJwt -> {
+        credential.disclosures.entries.associate {
```

Comment:
better use the JSON object ... e.g. if we don't have a disclosure for it (e.g. it can't be selectively disclosed, and is always in plain text)